### ISSUE TITLE
Setup-TraefikContainerForNavContainers.ps1: add param to exclude port mapping

### ISSUE THREAD

**Issue by ti-jalopez** [microsoft/navcontainerhelper#1573]
**New param to exclude port mapping or change mapping**
Please, add a new param in Setup-TrafikContainerForNavContainers: -ExcludePortsMapping, that allow to enter a list of ports for not include in  docker run -p.
I need to exclude port 8080 for mapping, because in the same server I'm using this port and I want to keep this behavior.
Alternatively, add a new param to redefine mappings. For example: -PortsCommand '-p 443:443 -p 80:80'.

**Scripts used to create container and cause the issue**
```
$publicDnsName = 'my.public.dom'
$publicLocalIP = '192.168.100.100'
$emailDnsACME = 'emal@mydomain.es'

Setup-TraefikContainerForBCContainers `
  -PublicDnsName $publicDnsName -IP $publicLocalIP -ContactEMailForLetsEncrypt $emailDnsACME `
  -overrideDefaultBinding -forceHttpWithTraefik
```
**Error output of script**
```
docker : docker: Error response from daemon: failed to create endpoint condescending_grothendieck on network nat: failed during hnsCallRawResponse: hnsCall failed in Win32: 
The process cannot access the file because it is being used by another process. (0x20).
En C:\Program Files\WindowsPowerShell\Modules\BcContainerHelper\1.0.17\ContainerHandling\Setup-TraefikContainerForNavContainers.ps1: 166 Carácter: 13
+             docker run -p 8080:8080 -p 443:443 -p 80:80 --restart alw ..
```

**Screenshots**

**Manual fix**
I see source code in https://github.com/microsoft/navcontainerhelper/blob/master/ContainerHandling/Setup-TraefikContainerForNavContainers.ps1. Then, I prepare a script to create the new container based on it.
The new traefik container is created, without map port 8080, as I want. since I don't expose traefik admin website.
```
$traefikForBcBasePath = "C:\ProgramData\BcContainerHelper\traefikforbc"
$traefikDockerImage = "stefanscherer/traefik-windows:v1.7.12"
$traefikContainerName = "traefik"
docker run `
  -p 443:443 -p 80:80 --restart always --name $traefikContainerName --hostname $traefikContainerName -d `
  -v ((Join-Path $traefikForBcBasePath "config") + ":c:/etc/traefik") `
  -v \\.\pipe\docker_engine:\\.\pipe\docker_engine $traefikDockerImage --docker.endpoint=npipe:////./pipe/docker_engine
```

**Comment by freddydk** [comment]
In the next bccontainerhelper preview, there is a parameter called -doNotPublishAdminPort on the setup function.

**Comment by freddydk** [comment]
Shipped in 1.0.18

**Comment by ti-jalopez** [comment]
Hi @freddydk ,
I'm testing new param in version 1.0.18, but something is wrong in my tests

**Script**
`Setup-TraefikContainerForBCContainers   -PublicDnsName $publicDnsName -IP $publicLocalIP -ContactEMailForLetsEncrypt $emailDnsACME   -overrideDefaultBinding -doNotPublishAdminPort`

**Error output**
```
Traefik image already up to date
Running traefik
DockerDo : docker: invalid reference format.
See 'docker run --help'.
ExitCode: 125
Commandline: docker run -p 443:443 -p 80:80 --restart always -v ((Join-Path c:\programdata\bccontainerhelper\traefikforbc "config") + ":c:/etc/traefik") -v 
\\.\pipe\docker_engine:\\.\pipe\docker_engine --detach stefanscherer/traefik-windows:v1.7.12 --docker.endpoint=npipe:////./pipe/docker_engine
En C:\Program Files\WindowsPowerShell\Modules\BcContainerHelper\1.0.18\ContainerHandling\Setup-TraefikContainerForNavContainers.ps1: 177 Carácter: 9
+         DockerDo -command run -imageName "$traefikDockerImage --docke ...
+         ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : NotSpecified: (:) [Write-Error], WriteErrorException
    + FullyQualifiedErrorId : Microsoft.PowerShell.Commands.WriteErrorException,DockerDo

````
**Additional information**
With the new parameter -additionalParameters it throws the same error. See issue  #1574.
Tested in Windows 10 as docker version
```
Client: Docker Engine - Community
 Cloud integration: 1.0.4
 Version:           20.10.0
 API version:       1.41
 Go version:        go1.13.15
 Git commit:        7287ab3
 Built:             Tue Dec  8 18:55:31 2020
 OS/Arch:           windows/amd64
 Context:           default
 Experimental:      true

Server: Docker Engine - Community
 Engine:
  Version:          20.10.0
  API version:      1.41 (minimum version 1.24)
  Go version:       go1.13.15
  Git commit:       eeddea2
  Built:            Tue Dec  8 19:07:44 2020
  OS/Arch:          windows/amd64
  Experimental:     false
```
and Windows Server 2019:
```
C:\Windows\system32>docker version
Client: Mirantis Container Runtime
 Version:           19.03.13
 API version:       1.40
 Go version:        go1.13.15
 Git commit:        0c38b2d
 Built:             11/12/2020 21:33:08
 OS/Arch:           windows/amd64
 Experimental:      false

Server: Mirantis Container Runtime
 Engine:
  Version:          19.03.13
  API version:      1.40 (minimum version 1.24)
  Go version:       go1.13.15
  Git commit:       5b5efb2d49
  Built:            11/12/2020 21:31:44
  OS/Arch:          windows/amd64
  Experimental:     false

```

**Comment by freddydk** [comment]
Could you try to add -isolation hyperv

**Comment by freddydk** [comment]
Not needed - I think I see the error

**Comment by freddydk** [comment]
Could you open this file in PowerShell ISE:
```
C:\Program Files\WindowsPowerShell\Modules\BcContainerHelper\1.0.18\ContainerHandling\Setup-TraefikContainerForNavContainers.ps1
```
and modify line 167 to
```
        $parameters = @("-p 443:443", "-p 80:80", "--restart always", "-v ($(Join-Path $traefikForBcBasePath ""config"") + "":c:/etc/traefik"")", "-v \\.\pipe\docker_engine:\\.\pipe\docker_engine")
```
and then restart PowerShell and retry?

**Comment by freddydk** [comment]
just updated the fix

**Comment by ti-jalopez** [comment]
Great, it run fine 👍 
Best regards,

**Comment by freddydk** [comment]
Will get this into 1.0.19 - sorry about the checkin - 1.0.18 was shipped as an emergency due to signing problems and this one wasn't tested enough.

**Comment by freddydk** [comment]
Shipped 1.0.19 with this fix
